input/postgres/relation_stats.go

```diff
@@ -156,3 +169,52 @@ func GetIndexStats(db *sql.DB, postgresVersion state.PostgresVersion, ignoreRege
 
 	return
 }
+
+func GetColumnStats(logger *util.Logger, db *sql.DB, globalCollectionOpts state.CollectionOpts, systemType string) (columnStats state.PostgresColumnStatsMap, err error) {
+	var sourceTable string
+
+	helperExists := false
+	db.QueryRow(QueryMarkerSQL + columnStatsHelperSQL).Scan(&helperExists)
+
+	if helperExists {
+		logger.PrintVerbose("Found pganalyze.get_column_stats() stats helper")
+		sourceTable = "pganalyze.get_column_stats()"
+	} else {
+		if systemType != "heroku" && !connectedAsSuperUser(db, systemType) && !connectedAsMonitoringRole(db) && globalCollectionOpts.TestRun {
+			logger.PrintInfo("Warning: You are not connecting as superuser. Please setup" +
+				" the monitoring helper functions (https://github.com/pganalyze/collector#setting-up-a-restricted-monitoring-user)" +
+				" or connect as superuser, to get table statistics for all roles.")
+		}
+		sourceTable = "pg_catalog.pg_stats"
+	}
+
+	stmt, err := db.Prepare(QueryMarkerSQL + fmt.Sprintf(columnStatsSQL, sourceTable))
+	if err != nil {
+		return nil, err
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.Query()
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var statsMap = make(state.PostgresColumnStatsMap)
+
+	for rows.Next() {
+		var s state.PostgresColumnStats
+		var key string
+
+		err := rows.Scan(
+			&s.SchemaName, &s.TableName, &s.ColumnName, &s.Inherited, &s.NullFrac, &s.AvgWidth, &s.NDistinct, &s.Correlation)
+		if err != nil {
+			return nil, err
+		}
+
+		key = s.SchemaName + s.TableName + s.ColumnName
```

Comment:
This isn't safe if both table `foo`, column `barbaz` and table `foobar`, column `baz` exist, I think. That's probably not very likely, but maybe it's likely enough that we should handle it better. It's a shame Go makes multilevel maps a pain--can't think of a super elegant solution here.

Reply:
I'll update the key to use `\t` as a delimiter for now. Unless @lfittl thinks a nested map is needed?

Reply:
Commented on this separately - I think we should match the pattern we have for statements (use a struct as the key)